src/js/components/Baani/Baani.js

```diff
@@ -495,154 +498,154 @@ class Baani extends React.PureComponent {
+                  </div>
+                ))}
+              </div>
+              {/* Spanish Translation Languages */}
+              <div
+                className={`${mixedViewBaaniClass}-translation ${paragraphModeClass}`}
+              >
                   {
                   spanishTranslationLanguage && (
-                    <>
-                      <div
-                        key={SPANISH_LANGUAGE}
-                        className={`${mixedViewBaaniClass}-translation-${SPANISH_LANGUAGE} ${paragraphModeClass}`}
-                      >
-                        {shabads.map((shabad) =>
-                          this.createShabadLine(
-                            shabad,
-                            this.getTranslationForLanguage(shabad, SPANISH_LANGUAGE)
-                          )
-                        )}
-                      </div>                      
-                    </>
-                  )}
-                </div>
-                {/* Punjabi Translation Languages */}
-                <div
-                  className={`${mixedViewBaaniClass}-steek ${paragraphModeClass}`}
-                >
-                  {steekLanguages.map((language) => (
-                    <div
-                      key={language}
-                      className={`${mixedViewBaaniClass}-steek-${language} ${paragraphModeClass}`}
-                    >
-                      {shabads.map((shabad) =>
-                        this.createShabadLine(
-                          shabad,
-                          this.getSteekForLanguage(shabad, language)
-                        )
-                      )}
-                    </div>
-                  ))}
-                </div>
-                {/* Hindi Translation Languages */}
-                <div
-                  className={`${mixedViewBaaniClass}-translation ${paragraphModeClass}`}
-                >
-                  {hindiTranslationLanguages.map((source) => (
+                  <>
                     <div
-                      key={source}
-                      className={`${mixedViewBaaniClass}-translation-${HINDI_LANGUAGE} ${paragraphModeClass}`}
+                      key={SPANISH_LANGUAGE}
+                      className={`${mixedViewBaaniClass}-translation-${SPANISH_LANGUAGE} ${paragraphModeClass}`}
                     >
                       {shabads.map((shabad) =>
                         this.createShabadLine(
                           shabad,
-                          this.getTranslationFromLanguageSource(shabad, HINDI_LANGUAGE, source)
+                            this.getTranslationForLanguage(shabad, SPANISH_LANGUAGE)
                         )
                       )}
                     </div>
-                  ))}
-                </div>
-                
-                {isParagraphMode ? null : (
+                  </>
+                )}
+              </div>
+              {/* Punjabi Translation Languages */}
+              <div
```

Comment:
@JapnitS Lets create a small component named DisplayMixedViewBaani and use that here.